services/vmagent-config-update/models/config.go

```diff
@@ -0,0 +1,122 @@
+package models
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"gopkg.in/yaml.v2"
+)
+
+type ConfigOptions func(config *Config)
+
+func NewConfig(options ...ConfigOptions) *Config {
```

Comment:
This looks too complicated and also is missing the validation step for passed params. For example, validating target address.